sentry/src/db/campaign.rs

```diff
@@ -50,11 +48,90 @@ pub async fn fetch_campaign(
     Ok(row.as_ref().map(Campaign::from))
 }
 
+pub async fn get_campaigns_by_channel(
+    pool: &DbPool,
+    channel_id: &ChannelId,
+) -> Result<Vec<Campaign>, PoolError> {
+    let client = pool.get().await?;
+    let statement = client.prepare("SELECT id, channel, creator, budget, validators, title, pricing_bounds, event_submission, ad_units, targeting_rules, created, active_from, active_to FROM campaigns WHERE channel_id = $1").await?;
+
+    let rows = client.query(&statement, &[&channel_id]).await?;
+
+    let campaigns = rows.iter().map(Campaign::from).collect();
+
+    Ok(campaigns)
+}
+
+pub async fn update_campaign(
```

Comment:
Instead of passing both, we can just pass the Campaign and instead have a method on `ModifyCampaign` to `apply()` the changes to a `Campaign`. This way the database call function will be strictly for updating the database and not handling any additional logic regarding modifying a campaign (and will be reusable in other scenarios)